utils/io.py

```diff
@@ -36,9 +36,11 @@ def copy_config_file(config_file, out_path, new_fields):
     config_out_file.close()
 
 
-def load_checkpoint(model, checkpoint_path, use_cuda=False):
+def load_checkpoint(model, checkpoint_path, use_cuda=False, amp=None):
     state = torch.load(checkpoint_path, map_location=torch.device('cpu'))
     model.load_state_dict(state['model'])
+    if amp and 'amp' in state:
```

Comment:
whats if the model is saved without amp but someone likes to finetune the model using amp?

Reply:
this is about loading an existing model - if it was saved **without** amp, it cannot be loaded by definition. if someone finetunes **with** amp it can be stored **with** the dict afterwords. loading the amp dict just means the values like weights are correctly initialized. if you use low precision (FP16) for finetuning a full precision (FP32) model, it should work overall.